store/changeset.go

```diff
@@ -4,36 +4,41 @@ package store
 // track writes. Deletion can be denoted by a nil value or explicitly by the
 // Delete field.
 type KVPair struct {
-	Key      []byte
-	Value    []byte
-	StoreKey string // optional
+	Key   []byte
+	Value []byte
 }
 
 // Changeset defines a set of KVPair entries.
+// storeKey -> []KVPair
 type Changeset struct {
-	Pairs []KVPair
+	Pairs map[string][]KVPair
 }
 
-func NewChangeset(pairs ...KVPair) *Changeset {
+func NewChangeset(storeKey string, pairs ...KVPair) *Changeset {
```

Comment:
`NewChangeset` accepts a single storeKey and pairs for that storeKey. Instead, we should just accept the map directly.